resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -270,7 +271,7 @@ private[spark] object KubernetesConf {
         .replaceAll("-+", "-"),
       "",
       KUBERNETES_DNSNAME_MAX_LENGTH
-    )
+    ).stripPrefix("-").stripSuffix("-")
```

Comment:
Note for my self or others: k8s label allow `dashes (-), underscores (_), dots (.)`, we alread replaced `underscores (_)`, `dots (.)` by `dashes (-)` before, so strip "-" prefix/suffix is enough in here.

Below tests are also passed:
```
assert(KubernetesConf.getAppNameLabel(".hello.") === "hello")
assert(KubernetesConf.getAppNameLabel("_hello_") === "hello")
```